https://bugs.documentfoundation.org/show_bug.cgi?id=142442
https://bug-attachments.documentfoundation.org/attachment.cgi?id=172646

Workaround gcc10 issues such as:

ucbstorage.cxx:(.text+0x645b): undefined reference to `non-virtual thunk to cppu::WeakImplHelper<com::sun::star::io::XInputStream>::acquire()'

--- a/sot/source/sdstor/ucbstorage.cxx	2020-11-16 21:30:57.533122410 +0200
+++ b/sot/source/sdstor/ucbstorage.cxx	2020-11-17 13:16:01.689775499 +0200
@@ -78,6 +78,8 @@
 static int nOpenStreams=0;
 #endif
 
+template class css::uno::Reference < css::io::XInputStream >;
+
 typedef ::cppu::WeakImplHelper < XInputStream, XSeekable > FileInputStreamWrapper_Base;
 class FileStreamWrapper_Impl : public FileInputStreamWrapper_Base
 {
--- a/svtools/source/misc/embedhlp.cxx	2020-11-17 09:31:23.953935644 +0200
+++ b/svtools/source/misc/embedhlp.cxx	2020-11-17 13:11:45.966880354 +0200
@@ -56,6 +56,8 @@
 
 using namespace com::sun::star;
 
+template class uno::Reference< io::XInputStream > ;
+
 namespace svt {
 
 class EmbedEventListener_Impl : public ::cppu::WeakImplHelper < embed::XStateChangeListener,
--- a/svtools/source/misc/imageresourceaccess.cxx.orig  2020-12-15 10:08:58.963211392 +0000
+++ a/svtools/source/misc/imageresourceaccess.cxx   2020-12-15 10:10:27.089875761 +0000
@@ -31,6 +31,8 @@
 #include <unotools/streamwrap.hxx>
 #include <cppuhelper/implbase.hxx>

+template class com::sun::star::uno::Reference<com::sun::star::io::XInputStream>;
+
 namespace svt::GraphicAccess
 {

--- a/starmath/source/mathml/mathmlimport.cxx	2021-05-14 18:41:06.000000000 +0000
+++ b/starmath/source/mathml/mathmlimport.cxx	2021-05-31 09:57:07.911244058 +0000
@@ -80,6 +80,8 @@
 using namespace ::com::sun::star;
 using namespace ::xmloff::token;
 
+template class uno::Reference< io::XInputStream > ;
+
 namespace
 {
 std::unique_ptr<SmNode> popOrZero(SmNodeStack& rStack)
--- a/filter/source/msfilter/msdffimp.cxx	2020-11-17 13:24:36.752179106 +0200
+++ b/filter/source/msfilter/msdffimp.cxx	2020-11-17 13:24:53.035283815 +0200
@@ -187,6 +187,8 @@
 using namespace drawing             ;
 using namespace container           ;
 
+template class uno::Reference< io::XInputStream > ;
+
 // static counter for OLE-Objects
 static sal_uInt32 nMSOleObjCntr = 0;
 #define MSO_OLE_Obj "MSO_OLE_Obj"
--- a/writerfilter/source/rtftok/rtfdocumentimpl.cxx	2020-11-17 13:28:01.622843541 +0200
+++ b/writerfilter/source/rtftok/rtfdocumentimpl.cxx	2020-11-17 13:28:15.432664842 +0200
@@ -54,6 +54,8 @@
 
 using namespace com::sun::star;
 
+template class uno::Reference< io::XInputStream > ;
+
 namespace
 {
 /// Returns an util::DateTime from a 'YYYY. MM. DD.' string.
--- a/sd/source/filter/xml/sdxmlwrp.cxx	2020-11-17 13:30:37.404161089 +0200
+++ b/sd/source/filter/xml/sdxmlwrp.cxx	2020-11-17 13:30:48.510684039 +0200
@@ -83,6 +83,8 @@
 using namespace com::sun::star::document;
 using namespace comphelper;
 
+template class uno::Reference< io::XInputStream > ;
+
 #define SD_XML_READERROR ErrCode(1234)
 
 char const sXML_export_impress_meta_oasis_service[] = "com.sun.star.comp.Impress.XMLOasisMetaExporter";
--- a/sw/source/filter/xml/xmlimp.cxx	2020-12-15 15:45:19.806017075 +0000
+++ a/sw/source/filter/xml/xmlimp.cxx	2020-12-15 15:51:52.252673387 +0000
@@ -90,6 +90,8 @@
 using namespace ::xmloff::token;
 using namespace ::std;
 
+template class uno::Reference< io::XInputStream > ;
+
 namespace {
 
 enum SwXMLDocTokens
--- a/editeng/source/xml/xmltxtimp.cxx	2020-12-15 17:19:05.992536014 +0000
+++ b/editeng/source/xml/xmltxtimp.cxx	2020-12-15 17:22:54.112530121 +0000
@@ -45,6 +45,8 @@
 using namespace cppu;
 using namespace xmloff::token;
 
+template class ::cppu::WeakImplHelper< css::io::XInputStream >;
+
 namespace {
 
 class SvxXMLTextImportContext : public SvXMLImportContext
--- a/filter/source/msfilter/msoleexp.cxx	2020-12-15 19:33:11.738992484 +0000
+++ b/filter/source/msfilter/msoleexp.cxx	2020-12-15 19:36:50.725653378 +0000
@@ -47,6 +47,8 @@
 
 using namespace ::com::sun::star;
 
+template class uno::Reference< io::XOutputStream > ;
+
 static SvGlobalName GetEmbeddedVersion( const SvGlobalName& aAppName )
 {
     if ( aAppName == SvGlobalName( SO3_SM_CLASSID_60 ) )
--- a/editeng/source/misc/svxacorr.cxx	2020-12-15 19:29:45.692331240 +0000
+++ b/editeng/source/misc/svxacorr.cxx	2020-12-15 19:32:22.338993770 +0000
@@ -79,6 +79,8 @@
 using namespace ::xmloff::token;
 using namespace ::utl;
 
+template class uno::Reference< io::XOutputStream > ;
+
 namespace {
 
 enum class Flags {
--- a/sd/source/filter/sdpptwrp.cxx	2020-12-15 19:43:10.755643449 +0000
+++ b/sd/source/filter/sdpptwrp.cxx	2020-12-15 19:44:35.588974556 +0000
@@ -40,6 +40,9 @@
 using namespace ::com::sun::star::task;
 using namespace ::com::sun::star::frame;
 
+using namespace ::com::sun::star;
+template class uno::Reference< io::XInputStream > ;
+
 typedef sal_Bool ( *ExportPPTPointer )( const std::vector< css::beans::PropertyValue >&, tools::SvRef<SotStorage> const&,
                                              Reference< XModel > const &,
                                              Reference< XStatusIndicator > const &,
--- a/sc/source/ui/app/drwtrans.cxx	2020-12-15 19:47:50.142302805 +0000
+++ b/sc/source/ui/app/drwtrans.cxx	2020-12-15 19:48:28.612301794 +0000
@@ -60,6 +60,8 @@
 
 using namespace com::sun::star;
 
+template class uno::Reference< io::XOutputStream > ;
+
 constexpr sal_uInt32 SCDRAWTRANS_TYPE_EMBOBJ    = 1;
 constexpr sal_uInt32 SCDRAWTRANS_TYPE_DRAWMODEL = 2;
 constexpr sal_uInt32 SCDRAWTRANS_TYPE_DOCUMENT  = 3;
--- a/filter/source/xsltdialog/xmlfilterjar.cxx	2020-12-15 19:11:19.902360231 +0000
+++ a/filter/source/xsltdialog/xmlfilterjar.cxx	2020-12-15 19:55:13.948957817 +0000
@@ -59,6 +59,9 @@
 
 using ::rtl::Uri;
 
+template class uno::Reference< io::XInputStream > ;
+template class uno::Reference< io::XOutputStream > ;
+
 XMLFilterJarHelper::XMLFilterJarHelper( const Reference< XComponentContext >& rxContext )
 : mxContext( rxContext ),
     sVndSunStarPackage( "vnd.sun.star.Package:" ),
--- a/sc/source/filter/xml/xmlimprt.cxx	2021-05-21 17:33:56.169361496 +0000
+++ b/sc/source/filter/xml/xmlimprt.cxx	2021-05-21 17:36:23.266034615 +0000
@@ -112,6 +112,8 @@
 using namespace ::xmloff::token;
 using namespace ::formula;
 
+template class uno::Reference< io::XInputStream > ;
+
 extern "C" SAL_DLLPUBLIC_EXPORT css::uno::XInterface*
 Calc_XMLOasisImporter_get_implementation(
     css::uno::XComponentContext* context, css::uno::Sequence<css::uno::Any> const& )
